frontend/web/src/features/bookingengine/listingdetails/views/rulesContainer.js

```diff
@@ -15,10 +15,17 @@ const RulesContainer = ({ rules, checkIn }) => {
     return `${attributes} ${allowedText}`;
   };
 
-  const formatHour = (hour) => {
-    const padded = hour.toString().padStart(2, "0");
-    return `${padded}:00`;
-  };
+ const formatHour = (value) => {
+  if (!value) return "";
+
+  const parts = value.toString().split(":");
+  
+  const hour = parts[0] || "00";
+  const minute = parts[1] || "00";
+
+  return `${hour.padStart(2, "0")}:${minute.padStart(2, "0")}`;
+};
+
```

Comment:
The whitelines do not contribute to better readability here so keep it as compact as possible.
```suggestion
 const formatHour = (value) => {
  if (!value) return "";
  const parts = value.toString().split(":");
  const hour = parts[0] || "00";
  const minute = parts[1] || "00";
  return `${hour.padStart(2, "0")}:${minute.padStart(2, "0")}`;
};
```